### PR TITLE
feat: add openscad_lsp

### PR DESCRIPTION
**Pull request by williamboman**
This server is a fork of sorts of the `openscad_ls` server that is
already supported by nvim-lspconfig. I have no insight into either
projects, but the `openscad-lsp` Cargo package seems to be more
actively maintained, see:

- `openscad_ls`: https://github.com/dzhu/openscad-language-server
- `openscad_lsp (this pr)`: https://github.com/Leathong/openscad-LSP

### CODE REVIEW

File: lua/lspconfig/server_configurations/openscad_lsp.lua [hunk 1]
```diff
@@ -0,0 +1,32 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'openscad-lsp' },
```
Comment (williamboman) [inline lua/lspconfig/server_configurations/openscad_lsp.lua:5]: One sec, need to verify this cmd (the binary name is correct but it seems to default to TCP transport).

Comment (williamboman) [reply lua/lspconfig/server_configurations/openscad_lsp.lua:5]: Yep, missing `--stdio` flag.

```suggestion
    cmd = { 'openscad-lsp', '--stdio' },
```

**Comment by calexanderberg** [comment]
I believe as of [April 2022](https://github.com/neovim/neovim/commit/6e6f5a783333d1bf9d6c719c896e72ac82e1ae54) openscad files are recognized, but I could be wrong.

**Comment by williamboman** [comment]
> I believe as of [April 2022](https://github.com/neovim/neovim/commit/6e6f5a783333d1bf9d6c719c896e72ac82e1ae54) openscad files are recognized and have a filetype pattern, however maybe I just misunderstood the issue here.

Ah, cool! Removed that section from the docs.

**Pull request merged**